src/upload-lib.ts

```diff
@@ -356,18 +356,16 @@ async function uploadPayload(
 ): Promise<string> {
   logger.info("Uploading results");
 
-  // If in test mode we don't want to upload the results
-  if (util.isInTestMode()) {
+  // If in test mode we don't want to upload the results,
+  if (util.shouldSkipSarifUpload()) {
     const payloadSaveFile = path.join(
       actionsUtil.getTemporaryDirectory(),
       "payload.json",
     );
-    logger.info(
-      `In test mode. Results are not uploaded. Saving to ${payloadSaveFile}`,
-    );
+    logger.info(`SARIF upload disabled. Saving to ${payloadSaveFile}`);
```

Comment:
Minor: This no longer explains why the SARIF upload is disabled. A simple fix might be wording it like this:

```suggestion
    logger.info(`SARIF upload disabled with an environment variable. Saving to ${payloadSaveFile}`);
```

A better fix would be to include the responsible environment variable, perhaps with:

```suggestion
    logger.info(`SARIF upload disabled by `${ isInTestMode() ? EnvVar.TEST_MODE : EnvVar.SKIP_SARIF_UPLOAD }`. Saving to ${payloadSaveFile}`);
```